javav2/example_code/ssm/src/main/java/com/example/scenario/SSMActions.java

```diff
@@ -0,0 +1,623 @@
+import software.amazon.awssdk.services.ssm.model.DocumentAlreadyExistsException;
+import software.amazon.awssdk.services.ssm.model.DocumentType;
+import software.amazon.awssdk.services.ssm.model.GetCommandInvocationRequest;
+import software.amazon.awssdk.services.ssm.model.GetCommandInvocationResponse;
+import software.amazon.awssdk.services.ssm.model.ListCommandInvocationsRequest;
+import software.amazon.awssdk.services.ssm.model.ListCommandInvocationsResponse;
+import software.amazon.awssdk.services.ssm.model.MaintenanceWindowFilter;
+import software.amazon.awssdk.services.ssm.model.MaintenanceWindowIdentity;
+import software.amazon.awssdk.services.ssm.model.OpsItem;
+import software.amazon.awssdk.services.ssm.model.OpsItemDataValue;
+import software.amazon.awssdk.services.ssm.model.OpsItemFilter;
+import software.amazon.awssdk.services.ssm.model.OpsItemFilterKey;
+import software.amazon.awssdk.services.ssm.model.OpsItemFilterOperator;
+import software.amazon.awssdk.services.ssm.model.OpsItemStatus;
+import software.amazon.awssdk.services.ssm.model.OpsItemSummary;
+import software.amazon.awssdk.services.ssm.model.SendCommandRequest;
+import software.amazon.awssdk.services.ssm.model.SendCommandResponse;
+import software.amazon.awssdk.services.ssm.model.SsmException;
+import software.amazon.awssdk.services.ssm.model.UpdateMaintenanceWindowRequest;
+import software.amazon.awssdk.services.ssm.model.UpdateMaintenanceWindowResponse;
+import software.amazon.awssdk.services.ssm.model.CreateOpsItemResponse;
+import software.amazon.awssdk.services.ssm.model.UpdateOpsItemRequest;
+import software.amazon.awssdk.services.ssm.model.GetOpsItemRequest;
+import software.amazon.awssdk.services.ssm.model.GetOpsItemResponse;
+import java.time.Duration;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+// snippet-start:[ssm.java2.actions.main]
+public class SSMActions {
+
+    private static SsmAsyncClient ssmAsyncClient;
+
+    private static SsmAsyncClient getAsyncClient() {
+        SdkAsyncHttpClient httpClient = NettyNioAsyncHttpClient.builder()
+            .maxConcurrency(100)
+            .connectionTimeout(Duration.ofSeconds(60))
+            .readTimeout(Duration.ofSeconds(60))
+            .writeTimeout(Duration.ofSeconds(60))
+            .build();
+
+        ClientOverrideConfiguration overrideConfig = ClientOverrideConfiguration.builder()
+            .apiCallTimeout(Duration.ofMinutes(2))
+            .apiCallAttemptTimeout(Duration.ofSeconds(90))
+            .retryPolicy(RetryPolicy.builder()
+                .numRetries(3)
+                .build())
+            .build();
+
+        if (ssmAsyncClient == null) {
```

Comment:
Shouldn't this if statement cover the builders above as well? You're running 3 builders before you check whether you needed to or not.